Bugsnag/Configuration/BugsnagConfiguration.m

```diff
@@ -92,7 +92,7 @@ - (nonnull id)copyWithZone:(nullable NSZone *)zone {
     [copy setMaxPersistedEvents:self.maxPersistedEvents];
     [copy setMaxPersistedSessions:self.maxPersistedSessions];
     [copy setMaxBreadcrumbs:self.maxBreadcrumbs];
-    copy->_metadata = [[BugsnagMetadata alloc] initWithDictionary:[[self.metadata toDictionary] mutableCopy]];
+    copy->_metadata = [[BugsnagMetadata alloc] initWithDictionary:[[self.metadata toDictionary] mutableCopy]]; //!OCLINT
```

Comment:
Is this being triggered because of a direct ivar access?

Reply:
Yes; fixing it "properly" would require a bigger change

Reply:
I haven't looked deeply, but wouldn't this be a matter of an anonymous `BugsnagMetadata` category that grants write access to the property?

Maybe these things that require bigger changes could be put as a ticket so we don't lose track?

Reply:
Because `BugsnagConfiguration+Private.h` contains the class extension (`@interface BugsnagConfiguration ()`) it's not possible to override that declaration in `BugsnagConfiguration.m`; the compiler will bail with `error: property has a previous declaration`

I think the clean fix it to add an `initWithConfiguration:` method that initializes by copying, and make `copyWithZone:` call into that.

Reply:
Or simply implement `setMetadata:` 😅